### PR TITLE
Add Documentation for Package Plugin

### PR DESCRIPTION
**Pull request by fendor**
Not as in-depth as I wished, though.

### CODE REVIEW

File: src/Haskell/Ide/Engine/Plugin/Package.hs [hunk 1]
```diff
@@ -83,30 +98,49 @@ addCmd = CmdSync $ \(AddParams rootDir modulePath pkg) -> do
...
       liftToGhc $ editHpackPackage absFp relModulePath pkg
     NoPackage -> return $ IdeResultFail (IdeError PluginError "No package.yaml or .cabal found" Null)
 
-data PackageType = CabalPackage FilePath
-                 | HpackPackage FilePath
-                 | NoPackage
-
+data PackageType = CabalPackage FilePath -- ^ Location of Cabal File.
+                 | HpackPackage FilePath -- ^ Location of `package.yaml`
+                 | NoPackage -- ^ No package format ahs been found.
```
Comment (alanz) [inline src/Haskell/Ide/Engine/Plugin/Package.hs]: typo, 'ahs'

File: src/Haskell/Ide/Engine/Plugin/Package.hs [hunk 2]
```diff
@@ -64,14 +64,29 @@ packageDescriptor plId = PluginDescriptor
...
-  , fileParam      :: FilePath -- ^ A path to a module inside the
-                               -- library/executable/test-suite you want to
-                               -- add the package to.
-  , packageParam   :: T.Text   -- ^ The name of the package to add.
+  { rootDirParam   :: FilePath   -- ^ The root directory.
+  , fileParam      :: ModulePath -- ^ A path to a module inside the
+                                 -- library/executable/test-suite you want to
+                                 -- add the package to. May be a realtive oir
+                                 -- absolute path, thus, must be normalised.
+  , packageParam   :: T.Text     -- ^ The name of the package to add.
```
Comment (alanz) [inline src/Haskell/Ide/Engine/Plugin/Package.hs]: Should this not be of type `Package`?

Comment (fendor) [reply src/Haskell/Ide/Engine/Plugin/Package.hs]: Absolutely!

File: src/Haskell/Ide/Engine/Plugin/Package.hs [hunk 3]
```diff
@@ -227,18 +267,26 @@ editCabalPackage file modulePath pkgName fileMap = do
...
+    -- | Add the given package name to the cabal file.
+    -- Package is appended to the dependency list.
+    addDep :: L.HasBuildInfo a => Package -> a -> a
     addDep dep x = L.buildInfo . L.targetBuildDepends .~ newDeps $ x
       where oldDeps = x ^. L.buildInfo . L.targetBuildDepends
             -- Add it to the bottom of the dependencies list
-            newDeps = oldDeps ++ [Dependency (mkPackageName dep) anyVersion]
+            -- TODO: we could sort the depencies and then insert it,
+            -- or insert it in order iff the list is already sorted.
+            newDeps = oldDeps ++ [Dependency (mkPackageName (T.unpack dep)) anyVersion]
```
Comment (fendor) [inline src/Haskell/Ide/Engine/Plugin/Package.hs]: This is the only code change to unify the API of package.yaml and *.cabal

**Pull request merged**